engine/console/gui.go

```diff
@@ -0,0 +1,99 @@
+package console
+
+import (
+	"fmt"
+	"github.com/gdamore/tcell/v2"
+	"github.com/rivo/tview"
+	"strconv"
+)
+
+type GatherUI struct {
+	app *tview.Application
+	flex *tview.Flex
+	myGatherPoint *tview.TextView
+	gameList *tview.Table
+	createBtn *tview.Button
+	newGame *tview.InputField
+	maxPlayers int
+	joinGames int
+}
+
+func checkNewGameField(textToCheck string, lastChar rune) bool {
+	_, err := strconv.Atoi(textToCheck)
+	if err != nil {
+		return false
+	}
+	return true
+}
+
+func GatherUIInit() *GatherUI {
```

Comment:
To follow go code convention, constructors should start with `New`. Rename it to `NewGatherUI`, please.